servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -1880,13 +1888,27 @@ void RenderForwardClustered::_render_scene(RenderDataRD *p_render_data, const Co
 		clear_color = p_default_bg_color;
 	}
 
+	RS::ViewportMSAA msaa = rb->get_msaa_3d();
+	bool use_msaa = msaa != RS::VIEWPORT_MSAA_DISABLED;
+
+	// Make sure RenderingEffects uses our msaa buffers
+	rb->set_rendering_to_msaa(use_msaa);
```

Comment:
This logic seems fishy to me. We read `msaa_3d` from the render buffers. Then check if `msaa != RS::VIEWPORT_MSAA_DISABLED` then use that to call a function and _and_ the result of that with the same check:
`p_rendering_to_msaa && (msaa_3d != RS::VIEWPORT_MSAA_DISABLED)`

Plus, we are allowing to get out of sync with the actual `rendering_to_msaa` state (or maybe that is desired?). 

I'm wondering if we should just let `set_rendering_to_msaa()` update the state directly without checking `msaa != RS::VIEWPORT_MSAA_DISABLED` again. Or is there another situation where this function might be called that needs it to operate like this?

Reply:
@clayjohn remember that there are various points where we can have effects, pre opaque, post opaque, pre transparent, post transparent, etc and even when MSAA is enabled, once resolved we now need to use the normal color buffers.

As someone might be writing an effect that can work whether MSAA is enabled or not, I figured that when they call one of the get helper functions, they would get the MSAA buffer but only if applicable. So while we're rendering to the MSAA buffers, we set `rendering_to_msaa` to true, resulting in the effect getting the MSAA buffers. Once we're finished and done and have run our final resolve, we set `rendering_to_msaa` to false resulting in the effect getting the resolved buffers.

But I wonder if I'm trying to help the effects writer too much and whether we should remove this all together, and leave it up to the effect writer to obtain the MSAA buffers when applicable.

Reply:
I've decided to undo this and simply add a `msaa` property to the getters. If true and if MSAA is enabled, the MSAA buffer is returned. This does leave the onus on the user to grab the correct buffer, and thus know when MSAA is enabled, whether it is appropriate to obtain the MSAA buffer or whether they should be using the resolved version, but I think that is acceptable especially seeing the primary use case of this is post effects, and they always run on resolved buffers.